.github/workflows/_buildpacks-release.yml

```diff
@@ -177,26 +201,103 @@ jobs:
       - name: Install Pack CLI
         uses: buildpacks/github-actions/setup-pack@v5.5.4
 
-      - name: Create Docker Image
-        run: pack buildpack package ${{ matrix.buildpack_id }} --config ${{ matrix.buildpack_output_dir }}/package.toml -v
+      - name: Check if version is already on Docker Hub
+        id: check
+        run: echo "published_to_docker=$(docker manifest inspect "${{ matrix.docker_repository }}:${{ matrix.buildpack_version }}" &> /dev/null && echo 'true' || echo 'false')" >> $GITHUB_OUTPUT
 
       - name: Login to Docker Hub
-        if: inputs.dry_run == false
+        if: steps.check.outputs.published_to_docker == 'false'
         uses: docker/login-action@v3.1.0
         with:
           registry: docker.io
           username: ${{ secrets.docker_hub_user }}
           password: ${{ secrets.docker_hub_token }}
 
-      - name: Check if version is already on Docker Hub
-        id: check
-        run: echo "published_to_docker=$(docker manifest inspect "${{ matrix.docker_repository }}:${{ matrix.buildpack_version }}" &> /dev/null && echo 'true' || echo 'false')" >> $GITHUB_OUTPUT
 
-      - name: Tag and publish buildpack
-        if: inputs.dry_run == false && steps.check.outputs.published_to_docker == 'false'
+      - name: Publish to Docker Hub
+        if: steps.check.outputs.published_to_docker == 'false'
+        env:
+          TARGETS: ${{ toJSON(matrix.targets) }}
         run: |
-          docker tag ${{ matrix.buildpack_id }} ${{ matrix.docker_repository }}:${{ matrix.buildpack_version }}
-          docker push ${{ matrix.docker_repository }}:${{ matrix.buildpack_version }}
+          temp_tag="${{ matrix.docker_repository }}:_${{ github.run_id }}"
```

Comment:
One thing I tried to use the Rust-based commands in this repo for was to offload any sufficiently complex script into something that could be easier to maintain and test.  This seems like good candidates for a new custom command and, if you were to do that, the subsequent temp tag cleanup could probably be rolled into that as well.

Reply:
Yeah, that's a good point. Though, I'm not super eager to rewrite it because:

1) I don't think the lifetime of this code is very long. When CNB support for multiarch launches and libcnb.rs supports that, a good chunk of this will be handled by `pack` and `cargo libcnb package`. 
2) There's a half a dozen or more command line calls to model in rust.

Though I think I'll take a stab at moving a few more calculations (like say `perm_tag` / `temp_tag`) into the rust matrix.